File: 0078-subsets/0078-subsets.cpp
class Solution {
public:
    vector<vector<int>>ans;
    void subset(vector<int>& nums,int i,int n,vector<int>& v){
        if(i==n){
            ans.push_back(v);
            return;
        }
        v.push_back(nums[i]);
        subset(nums,i+1,n,v);
        v.pop_back();
        subset(nums,i+1,n,v);
    }
    vector<vector<int>> subsets(vector<int>& nums) {
        int n=nums.size();
        vector<int>v;
        subset(nums,0,n,v);
        return ans;
    }
};

File: 0078-subsets/0078-subsets.java
class Solution {
    public List<List<Integer>> subsets(int[] nums) {
        int n=nums.length;
        List<List<Integer>> ans=new ArrayList<>();
        int subset=1<<n;

        for(int mask=0;mask<subset;mask++){
            List<Integer> temp=new ArrayList<>();
            for(int i=0;i<n;i++){
                if((mask & (1<<i))!=0){
                    temp.add(nums[i]);
                }
            }
            ans.add(temp);
        }
        return ans;
    }
}